src/hooks/useChangeNetwork.ts

```diff
@@ -0,0 +1,129 @@
+  const notify = useNotify()
+
+  const changeNetworkHandler = useCallback(
+    (desiredChainId: ChainId, successCallback?: () => void) => {
+      dispatch(updateChainId(desiredChainId))
+      successCallback?.()
+    },
+    [dispatch],
+  )
+
+  const changeNetwork = useCallback(
+    async (desiredChainId: ChainId, successCallback?: () => void, failureCallback?: () => void) => {
+      if (desiredChainId === chainId) return
+
+      const wallet = walletKey && SUPPORTED_WALLETS[walletKey]
+      if (wallet && isEVMWallet(wallet) && !isSolana(desiredChainId)) {
+        tryActivationEVM(wallet.connector)
+      }
+      if (wallet && isSolanaWallet(wallet) && !isEVM(desiredChainId)) {
+        tryActivationSolana(wallet.adapter)
+      }
+      if (isEVM(desiredChainId)) {
+        const switchNetworkParams = {
+          chainId: '0x' + Number(desiredChainId).toString(16),
+        }
+        const isWrongNetwork = error instanceof UnsupportedChainIdError
+        // If not connected EVM wallet, or connected EVM wallet and want to switch back to EVM network
+        if (
+          (!walletEVM.isConnected && !isWrongNetwork) ||
+          (walletEVM.isConnected && walletEVM.chainId === desiredChainId)
+        ) {
+          changeNetworkHandler(desiredChainId, successCallback)
+          return
+        }
+
+        const activeProvider = library?.provider ?? window.ethereum
+        if (activeProvider && activeProvider.request) {
+          try {
+            await activeProvider.request({
+              method: 'wallet_switchEthereumChain',
+              params: [switchNetworkParams],
+            })
+            successCallback?.()
+          } catch (switchError) {
+            // This is a workaround solution for Coin98
+            const isSwitchError =
+              typeof switchError === 'object' && switchError && Object.keys(switchError)?.length === 0
+            // This error code indicates that the chain has not been added to MetaMask.
+            if (switchError?.code === 4902 || switchError?.code === -32603 || isSwitchError) {
+              const addNetworkParams = getEVMAddNetworkParams(desiredChainId)
+              const value = await activeProvider.request({
+                method: 'wallet_addEthereumChain',
+                params: [addNetworkParams],
+              })
+              // wallet_addEthereumChain method return null mean it successful
+              if (value === null) {
+                successCallback?.()
+              } else {
+                notify({
+                  title: t`Failed to switch network`,
+                  type: NotificationType.ERROR,
+                  summary: t`In order to use KyberSwap on ${NETWORKS_INFO[desiredChainId].name}, you must accept the network in your wallet.`,
+                })
+                failureCallback?.()
+              }
+            } else {
+              // handle other "switch" errors
+              failureCallback?.()
+              notify({
```

Comment:
duplicate code, let do like branch main
https://github.com/KyberNetwork/kyberswap-interface/blob/main/src/hooks/useActiveNetwork.ts#L46

Reply:
They have diferent message. If we make `showError` accept param `summary`, it not making code much shorter, so I prefer keep it as is